### PR TITLE
PLANNER-2187 Use java 11 minimum for OptaPlanner 8

### PR DESCRIPTION
**Pull request by rsynek**
### JIRA

https://issues.redhat.com/browse/PLANNER-2187

### Referenced pull requests
https://github.com/kiegroup/optaplanner-quickstarts/pull/6
https://github.com/kiegroup/optaplanner-website/pull/285

### Checklist
- [x] Documentation updated if applicable.
- [x] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by triceo — CHANGES_REQUESTED** [review]
As far as I know Kogito and Quarks still support Java 8. In my opinion, we should not do this before them, and if we do, I believe we will eventually be asked to revert it.

@ge0ffrey This is your call.

File: optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/DefaultPlannerBenchmark.java
Comment (ge0ffrey) [inline optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/DefaultPlannerBenchmark.java:143]: Nice one that this is cleaned up!!

File: optaplanner-build-parent/pom.xml
Comment (ge0ffrey) [inline optaplanner-build-parent/pom.xml:364]: Overwrite property `<version.dependency.plugin>` instead, just like we have this property:


    <version.surefire.plugin>2.22.2</version.surefire.plugin>

**Comment by ge0ffrey** [comment]
> As far as I know Kogito and Quarks still support Java 8. In my opinion, we should not do this before them, and if we do, I believe we will eventually be asked to revert it.
> 
> @ge0ffrey This is your call.

Kogito is also dropping Java 8 support for Kogito 1.0 AFAIK.
code.quarkus.io doesn't support java 8 no more. If you use JDK 8 with quarkus, it shows **warnings** in the console.

**Review by ge0ffrey — APPROVED** [review]

File: .github/workflows/jpa_integration_ci.yml
Comment (ge0ffrey) [inline .github/workflows/jpa_integration_ci.yml:9]: How do this work?
Should the master branch's file mention all branches?
Or should each branch's file mention its own branch?

**Review by triceo — APPROVED** [review]
I was not aware of the Kogito decision. If they are indeed switching to 11 for their 1.0, I agree we should follow suit.
I was merely objecting to us being the only ones to do this.

That said, won't this cause Jenkins trouble? AFAIK Jenkins still builds with Java 8. (Hence the loss of Sonar.)

File: .github/workflows/jpa_integration_ci.yml
Comment (rsynek) [reply .github/workflows/jpa_integration_ci.yml:9]: Each branch should have its own .yaml file(s) with a filter for that particular branch. I think the multiple branches mentioned in the .yaml file on the master branch is a shortcut if you want to build multiple branches using the same workflow.

I would rather keep the branches separated. This test repo proves that triggering the CI on branches with different workflow definitions works: https://github.com/rsynek/github-actions-test/actions

File: optaplanner-build-parent/pom.xml
Comment (rsynek) [reply optaplanner-build-parent/pom.xml]: Good catch, thanks!

**Comment by rsynek** [comment]
Jenkins build fails as it still runs Java 8. I need to figure out how to change it before we merge this.

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
I love the green color of all checks passing once again.

**Pull request merged**
